# Impacto de la pandemia del COVID19 en los estudiantes
* Márquez Estrada Arath Patricio
* Rivera Vargas Juan
* Briceño Díaz Sofía
* Benitez Garcia Saul Enrique
* Enríquez López José Andrés
* Juárez Fonseca César Eduardo



## Introducción

La crisis sanitaria provocada por la pandemia ha desencadenado una serie de cambios en varios sectores de la población, siendo el confinamiento uno que ha impactado de manera significativa en los estudiantes, lo cual ha modificado sus métodos de aprendizaje, sus hábitos, su rutina y su estilo de vida en general. Estos cambios en el estilo de vida puede desembocar en cambios en la salud física y mental de los estudiantes. Al ser un tema que nos afecta directamente a nosotros, y al ser una situación actual que se vive en todo el mundo, nos pareció una problemática interesante para analizar.

Consideramos de gran importancia el tener conocimiento de estos cambios debido a la inminente incertidumbre que existe respecto a la duración de esta pandemia o a la aparición de nuevos virus provocadas por la invasión desregulada de hábitats silvestres y el calentamiento global. Teniendo eso en mente, podemos tener una mejor respuesta ante este tipo de situaciones y poder reducir sus impactos negativos.

Además de ser un tema de salud pública, en términos comerciales, les puede ser de utilidad a las empresas el impacto que ha tenido la pandemia respecto al comportamiento de consumo de los estudiantes y poder identificar patrones para esgrimirse a partir de estos y diseñar estrategias de marketing.

Para analizar este tema, trabajaremos con un dataset que ha recopilado datos sobre los estudiantes durante esta pandemia. Entre las preguntas que buscamos responder con nuestros datos se encuentran:
* ¿Cómo ha afectado la pandemia a la salud mental de los estudiantes?
* ¿Ha habido cambios en la salud física de los estudiantes?
* ¿Cómo han cambiado los hábitos de estudio entre los estudiantes?
* ¿Qué relación existe entre la percepción de las clases en línea y el medio que utilizan para tomar las clases?
* ¿Cómo es la calidad del sueño de los estudiantes durante la pandemia?
* ¿Cómo es percibida la calidad de las clases en línea en general?
* ¿En qué actividad pasan más tiempo del día?
* ¿Se han visto afectadas las relaciones de los estudiantes durante la pandemia?


## 1. Librerías
Empezamos importando las librerías que necesitamos, además de montar Drive, donde tenemos nuestro dataset.

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


## 2. Cargar los datos

A continuación cargamos nuestro dataset y lo almacenamos en una variable llamada df como DataFrame.

In [ ]:
#df = pd.read_csv('/content/drive/MyDrive/BEDU/AnxietyPTSD-Data.csv', encoding='cp1252')
df = pd.read_csv('/content/drive/MyDrive/BEDU/COVID-19 Survey Student Responses.csv')

## 3. Análisis exploratorio de datos

Hacemos un análisis introductorio de nuestro dataset para familiarizarnos con los datos que vamos a trabajar. Para esto, observamos algunos de los datos de nuestro dataset, vemos el tamaño y estructura de nuestro DataFrame, incluyendo el nombre de las columnas y el tipo de datos que contienen.

In [ ]:
df.head()

ID  ... What you miss the most
0  R1  ...         School/college
1  R2  ...  Roaming around freely
2  R3  ...             Travelling
3  R4  ...    Friends , relatives
4  R5  ...             Travelling

[5 rows x 19 columns]

In [ ]:
df.tail()

ID  ... What you miss the most
1177  R1191  ...             Travelling
1178  R1192  ...    Friends , relatives
1179  R1193  ...         School/college
1180  R1194  ...         School/college
1181  R1195  ...         School/college

[5 rows x 19 columns]

In [ ]:
df.size

22458

In [ ]:
df.dtypes

ID                                                                                     object
Region of residence                                                                    object
Age of Subject                                                                          int64
Time spent on Online Class                                                            float64
Rating of Online Class experience                                                      object
Medium for online class                                                                object
Time spent on self study                                                              float64
Time spent on fitness                                                                 float64
Time spent on sleep                                                                   float64
Time spent on social media                                                            float64
Prefered social media platform                              

In [ ]:
df.columns

Index(['ID', 'Region of residence', 'Age of Subject',
       'Time spent on Online Class', 'Rating of Online Class experience',
       'Medium for online class', 'Time spent on self study',
       'Time spent on fitness', 'Time spent on sleep',
       'Time spent on social media', 'Prefered social media platform',
       'Time spent on TV', 'Number of meals per day', 'Change in your weight',
       'Health issue during lockdown', 'Stress busters', 'Time utilized',
       'Do you find yourself more connected with your family, close friends , relatives  ?',
       'What you miss the most'],
      dtype='object')

In [ ]:
df['Stress busters']

0                              Cooking
1       Scrolling through social media
2                   Listening to music
3                  Watching web series
4                         Social Media
                     ...              
1177                           Dancing
1178                Listening to music
1179                     Online gaming
1180                     Reading books
1181                           Talking
Name: Stress busters, Length: 1182, dtype: object

In [ ]:
df.shape

(1182, 19)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1182 entries, 0 to 1181
Data columns (total 19 columns):
 #   Column                                                                              Non-Null Count  Dtype  
---  ------                                                                              --------------  -----  
 0   ID                                                                                  1182 non-null   object 
 1   Region of residence                                                                 1182 non-null   object 
 2   Age of Subject                                                                      1182 non-null   int64  
 3   Time spent on Online Class                                                          1182 non-null   float64
 4   Rating of Online Class experience                                                   1158 non-null   object 
 5   Medium for online class                                                             1131 non-null

Con las celdas anteriores nos damos cuenta de que varias de las columnas nos pueden ayudar a responder directamente algunas de nuestras preguntas. El dataset contiene 1182 filas, lo cual nos puede dar una idea inicial de cómo ha afectado la pandemia a los estudiantes. Los nombres de las columnas son bastante intuitivos, y la mayoría parecen tener el tipo de datos correcto, con pocas excepciones, por ejemplo el tiempo pasado en TV, que en vez de un valor numérico tiene un tipo de datos object.

## 4. Limpieza de datos y agregaciones

In [ ]:
df.isna().sum()

ID                                                                                     0
Region of residence                                                                    0
Age of Subject                                                                         0
Time spent on Online Class                                                             0
Rating of Online Class experience                                                     24
Medium for online class                                                               51
Time spent on self study                                                               0
Time spent on fitness                                                                  0
Time spent on sleep                                                                    0
Time spent on social media                                                             0
Prefered social media platform                                                         0
Time spent on TV     

*Al parecer tenemos dos columnas con NaNs "**Rating of Online Class experience**" y "**Medium for online class**"* 

In [ ]:
df[df['Medium for online class'].isna()].head()

ID  ...         What you miss the most
32    R33  ...  Nothing this is my usual life
34    R35  ...                 Eating outside
39    R40  ...                     Travelling
66    R67  ...                 Eating outside
129  R130  ...                 Eating outside

[5 rows x 19 columns]

*En la columna **Medium for online class** sustituimos los NaNs por ***'No medium'*** para considerarlos en nuestro Dataset.*

In [ ]:
df['Medium for online class'] = df['Medium for online class'].fillna('No medium')

In [ ]:
df['Medium for online class']

0       Laptop/Desktop
1           Smartphone
2       Laptop/Desktop
3           Smartphone
4       Laptop/Desktop
             ...      
1177        Smartphone
1178        Smartphone
1179        Smartphone
1180    Laptop/Desktop
1181            Tablet
Name: Medium for online class, Length: 1182, dtype: object

*En la columna **Rating of Online Class experience** sustituimos los NaNs por ***'No rating'*** para considerarlos en nuestro Dataset.*

In [ ]:
df['Rating of Online Class experience'] = df['Rating of Online Class experience'].fillna('No rating')

In [ ]:
df.isna().sum()

ID                                                                                    0
Region of residence                                                                   0
Age of Subject                                                                        0
Time spent on Online Class                                                            0
Rating of Online Class experience                                                     0
Medium for online class                                                               0
Time spent on self study                                                              0
Time spent on fitness                                                                 0
Time spent on sleep                                                                   0
Time spent on social media                                                            0
Prefered social media platform                                                        0
Time spent on TV                

*Ya tenemos nuestro dataframe limpio para poder trabajar sobre él*


**Aplicaremos funciones de agregación para poder responder algunas preguntas anteriormente planteadas**

*Primero una descripción general de nuestro dataframe*



In [ ]:
df.describe()

Age of Subject  ...  Number of meals per day
count     1182.000000  ...              1182.000000
mean        20.165821  ...                 2.917936
std          5.516467  ...                 0.828698
min          7.000000  ...                 1.000000
25%         17.000000  ...                 2.000000
50%         20.000000  ...                 3.000000
75%         21.000000  ...                 3.000000
max         59.000000  ...                 8.000000

[8 rows x 7 columns]

*trataremos de dar respuesta a la pregunta* ***¿Cómo es percibida la calidad de las clases en línea en general?***

In [ ]:
df.groupby("Rating of Online Class experience")["ID"].count()

Rating of Online Class experience
Average      387
Excellent     98
Good         230
No rating     24
Poor          30
Very poor    413
Name: ID, dtype: int64

*De lo anterior podemos notar que la percepción general de las clases en línea pasa de lo regular a muy deficiente.*

***¿Qué relación existe entre la percepción de las clases en línea y el medio que utilizan para tomar las clases?***

In [ ]:
df.groupby(['Medium for online class', 'Rating of Online Class experience'])['ID'].count().unstack().fillna(0)

Rating of Online Class experience  Average  Excellent  ...  Poor  Very poor
Medium for online class                                ...                 
Any Gadget                             1.0        0.0  ...   0.0        1.0
Laptop/Desktop                       199.0       43.0  ...  11.0      182.0
No medium                             10.0        2.0  ...   0.0        8.0
Smartphone                           169.0       46.0  ...  18.0      211.0
Smartphone or Laptop/Desktop           1.0        2.0  ...   1.0        0.0
Tablet                                 7.0        5.0  ...   0.0       11.0

[6 rows x 6 columns]

Podemos apreciar que realmente no hay una diferencia significativa entre la perspectiva sobre la experiencia de las clases en linea respecto al medio que utilizan

## 5. Transformación, filtración y ordenamiento de datos

In [ ]:
set(df['Time spent on TV'])

{' ',
 '0',
 '0.1',
 '0.25',
 '0.3',
 '0.4',
 '0.5',
 '0.6',
 '0.75',
 '1',
 '1.5',
 '15',
 '2',
 '2.5',
 '3',
 '3.5',
 '4',
 '4.5',
 '5',
 '6',
 '7',
 '8',
 'N',
 'No tv',
 'n'}

In [ ]:
df['Time spent on TV'] = df['Time spent on TV'].replace({'N': 0, 'No tv':0, 'n': 0, ' ':0}).astype(float)

con la transformación de la columna de "Time spent on TV" ahora podemos responder a la pregunta.

***¿En qué actividad pasan más tiempo del día?***

In [ ]:
df[["Time spent on fitness","Time spent on social media",
    "Time spent on TV","Time spent on self study"]].agg("sum")


Time spent on fitness          905.20
Time spent on social media    2796.25
Time spent on TV              1198.50
Time spent on self study      3441.50
dtype: float64

El estudio autodidacto parece ser la actividad a la que le dedican más tiempo durante el día.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1182 entries, 0 to 1181
Data columns (total 19 columns):
 #   Column                                                                              Non-Null Count  Dtype  
---  ------                                                                              --------------  -----  
 0   ID                                                                                  1182 non-null   object 
 1   Region of residence                                                                 1182 non-null   object 
 2   Age of Subject                                                                      1182 non-null   int64  
 3   Time spent on Online Class                                                          1182 non-null   float64
 4   Rating of Online Class experience                                                   1182 non-null   object 
 5   Medium for online class                                                             1182 non-null

In [ ]:
for i in df.loc[: , df.dtypes==object].columns:
  print(set(df[i]))

{'R66', 'R282', 'R1047', 'R717', 'R322', 'R642', 'R9', 'R993', 'R707', 'R1095', 'R507', 'R910', 'R848', 'R999', 'R1129', 'R406', 'R393', 'R1010', 'R561', 'R346', 'R969', 'R1068', 'R1126', 'R1140', 'R451', 'R487', 'R238', 'R41', 'R584', 'R515', 'R857', 'R991', 'R419', 'R143', 'R1117', 'R1144', 'R91', 'R88', 'R383', 'R573', 'R877', 'R116', 'R586', 'R537', 'R818', 'R291', 'R46', 'R1089', 'R645', 'R351', 'R26', 'R771', 'R1054', 'R288', 'R147', 'R495', 'R202', 'R18', 'R685', 'R1113', 'R1037', 'R10', 'R653', 'R1143', 'R86', 'R382', 'R1056', 'R67', 'R331', 'R635', 'R76', 'R932', 'R96', 'R739', 'R958', 'R315', 'R1065', 'R1127', 'R332', 'R702', 'R172', 'R976', 'R741', 'R194', 'R226', 'R708', 'R943', 'R752', 'R588', 'R594', 'R704', 'R449', 'R851', 'R690', 'R1048', 'R534', 'R244', 'R471', 'R628', 'R845', 'R740', 'R678', 'R405', 'R78', 'R675', 'R962', 'R935', 'R1115', 'R334', 'R938', 'R16', 'R256', 'R1182', 'R110', 'R1190', 'R513', 'R263', 'R945', 'R442', 'R53', 'R355', 'R224', 'R591', 'R434', 'R5

Vemos que los datos en las columnas tienen una estructura adecuada. Hacemos ahora varios subconjuntos para responder algunas de las preguntas iniciales.

**¿Cómo es la calidad del sueño de los estudiantes durante la pandemia?**

In [ ]:
df_duerme_menos_8_horas = df[df['Time spent on sleep']<8]
df_duerme_menos_8_horas.ID.count()

487

La cantidad de sueño ideal es entre 8 y 10 horas. Vemos que hay 487 estudiantes de los encuestados que duermen menos de 8 horas. La falta de sueño puede traer problemas de salud a largo plazo.

**¿Ha habido cambios en la salud física de los estudiantes?**

In [ ]:
df_problema_de_salud = df[df['Health issue during lockdown']=='YES']
df_problema_de_salud.ID.count()

161

In [ ]:
df_cambios_de_peso =  df_duerme_menos_8_horas[df_duerme_menos_8_horas['Change in your weight']!='Remain Constant']


Hacemos otro subconjunto de datos de los estudiantes que han tenido problemas de salud durante el confinamiento, y observamos que hay **161** estudiantes que han tenido algun problema de salud. Además, de los estudiantes que duermen menos de 8 horas, creamos otro subconjunto incluyendo los que han tenido cambios en su peso.

**¿Se han visto afectadas las relaciones de los estudiantes durante la pandemia?**

In [ ]:
def misses_something(x):
  for i in x:
    i = i.lower()
    if i == 'nothing':
      return False
  return True


In [ ]:
df_peso_y_extrana_cosas = df_cambios_de_peso[df_cambios_de_peso['What you miss the most'].str.split().map(misses_something)]

In [ ]:
df_peso_extrana_desconectado = df_peso_y_extrana_cosas[df_peso_y_extrana_cosas['Do you find yourself more connected with your family, close friends , relatives  ?']=='NO']

Usando los subconjuntos anteriores, incluimos uno con los estudiantes que echan de menos algo, y que no se sienten más conectados con sus familias y amigos. Éstos parecen buenos candidatos de estudiantes que pueden desarrollar problemas de salud mental y física.